File: smooth_function.py
def smooth_function(original_pixel, lsb2, lsb1):
    # Convert to integer for bitwise operations
    original_pixel = int(original_pixel)
    
    # Set LSB1 (first bit)
    watermarked_pixel = original_pixel & ~1  # Clear first bit
    watermarked_pixel |= lsb1 & 1  # Set first bit to LSB1
    
    # Set LSB2 (second bit)
    watermarked_pixel &= ~2  # Clear second bit
    watermarked_pixel |= (lsb2 & 1) << 1  # Set second bit to LSB2
    
    return watermarked_pixel

File: smooth_function.m
function [Watermarked_Pixel] = smooth_function(Original_Pixel,LSB2,LSB1)   

Watermarked_Pixel = bitset ((bitset(Original_Pixel,1,LSB1,'uint8')),2,LSB2,'uint8');
% display(Watermarked_Pixel);

%     y = double(1*LSB1+2*LSB2+4*LSB3) - double(bitget(Original_Pixel,1)*1+bitget(Original_Pixel,2)*2+bitget(Original_Pixel,3)*4);

%     if abs(y) < 5
%         Watermarked_Pixel =  uint8(double(Original_Pixel) + y);
%     elseif y <= -5
%         Watermarked_Pixel =  uint8(double(Original_Pixel) + y + 8);
%     elseif y >= 5
%         if (Original_Pixel==0) && (y==5||y==6||y==7)
%             Watermarked_Pixel =  4*LSB3+2*LSB2+1*LSB1;  
%         elseif (Original_Pixel==1) && (y==5||y==6)
%             Watermarked_Pixel =  4*LSB3+2*LSB2+1*LSB1;
%         elseif (Original_Pixel==2) && (y==5)
%             Watermarked_Pixel =  4*LSB3+2*LSB2+1*LSB1;
%         else
%             Watermarked_Pixel =  uint8(double(Original_Pixel) + y - 8);
%         end
%         
%     end        
end

